src/main/java/uk/gov/hmcts/reform/slc/services/servicebus/MessageProcessor.java

```diff
@@ -34,29 +34,38 @@ public MessageProcessor(
     }
 
     public void process() {
-        try {
-            IMessageReceiver messageReceiver = receiverProvider.get();
-            Instant receiveStartTime = Instant.now();
+        logger.info("Processing messages from the queue...");
+
+        IMessageReceiver messageReceiver = null;
 
-            processNextMessage(messageReceiver, receiveStartTime);
+        try {
+            messageReceiver = receiverProvider.get();
+            IMessage message = receiveMessage(messageReceiver);
 
-            messageReceiver.close();
+            while (message != null) {
+                sendLetter(messageReceiver, message);
+                message = receiveMessage(messageReceiver);
+            }
 
+            logger.info("Finished processing queue");
         } catch (ConnectionException e) {
             insights.trackException(e);
-
             logger.error("Unable to connect to Service Bus", e);
-        } catch (ServiceBusException e) {
-            insights.trackException(e);
-
-            logger.error("Error closing connection");
+        } finally {
+            if (messageReceiver != null) {
+                try {
+                    messageReceiver.close();
```

Comment:
is message receiver instance of closable? can simplify try/catch by replacing try with resource.

Reply:
Unfortunately, this is Microsoft's ICloseable, not standard AutoClosable - can't do try-with-resources.

Reply:
aye, i think i tried myself. repetition is the mother of skill 👍